### PR TITLE
Hopefully a PR of all time?

### PR DESCRIPTION
**Pull request by macaquedev**
@myst-6 have fun :D

### CODE REVIEW

**Review by myst-6 — CHANGES_REQUESTED** [review]
I think the UI design is good!

Also, I think we should set `run_timeout` on the request to `1000`, and display a message if the code gets TLE. Although we can do that in a separate PR if you wish.

**Comment by macaquedev** [comment]
we can talk about TLE on discord, then I'll open a new PR depending on what we decide to do if that's OK.

**Comment by macaquedev** [comment]
oh no it doesn't work anymore

File: components/SCodeBlock/index.tsx [hunk 1]
```diff
@@ -93,10 +95,10 @@ export const SCodeBlock = ({ path }: SCodeBlockProps) => {
           "Content-Type": "application/json",
         },
         body: JSON.stringify({
-          language: languages[idx] ? languages[idx].pistonName : "python", // specify the language
-          files: [{"name": "sol." + (languages[idx] ? languages[idx].extension : "py"), "content": codes[idx]}], // specify the code to execute
+          language: languages[idx]!.pistonName, // specify the language
+          files: [{"name": "sol." + languages[idx]!.extension, "content": codes[idx]}], // specify the code to execute
           stdin: testCase, // specify the input
-          version: languages[idx] ? languages[idx].version : "3.10",
+          version: languages[idx]!,
```
Comment (myst-6) [inline components/SCodeBlock/index.tsx:101]: I think this is meant to be `languages[idx]!.version`.

Comment (macaquedev) [reply components/SCodeBlock/index.tsx]: yep also noticed that, moment of pure idiocy

**Review by myst-6 — APPROVED** [review]
orz! :)

**Pull request merged**